### PR TITLE
RCIAM-303_Show_whole_tree_for_cous_ui

### PR DESCRIPTION
**Pull request by lionick**
Show the whole tree of a cou, seperated by " / " (e.g parent / child / subchild) when adding person to a cou. 
Related issue : https://jira.argo.grnet.gr/browse/RCIAM-303

### CODE REVIEW

File: app/Model/Cou.php [hunk 1]
```diff
@@ -259,40 +259,73 @@ public function childCous($parentCou, $co_id, $includeParent=false) {
...
+      $ret[$child['Cou']['id']] = $parent_name . ':' . $child['Cou']['name'];
     }
-    
     return $ret;
   }
 
+  /**
+   * Get Parent COU From parentCouId.
+   *
+   * @since  COmanage Registry v3.1
```
Comment (ioigoume) [inline app/Model/Cou.php]: v3.1.1 instead of 3.1

File: app/Model/Cou.php [hunk 2]
```diff
@@ -259,40 +259,73 @@ public function childCous($parentCou, $co_id, $includeParent=false) {
...
-    $args['conditions']['Cou.id'] = $parentCouId;
-    $args['contain'] = false;
-    
-    $parent = $this->find('first', $args);
-    
-    if(!isset($parent['Cou']['id'])) {
-      throw new InvalidArgumentException(_txt('er.unknown', array($parentCouId)));
+  public function childCousById($parentCouId, $includeParent=false, $parent=NULL) {
+    // Find $parent if not exists
+    if(empty($parent)) {
```
Comment (ioigoume) [inline app/Model/Cou.php]: Why not even shorter?
$parent = empty($parent) ? $this->getParentCouById($parentCouId) : $parent;

File: app/Model/Cou.php [hunk 3]
```diff
@@ -259,40 +259,73 @@ public function childCous($parentCou, $co_id, $includeParent=false) {
...
     $children = $this->children($parentCouId,
                                 false,
-                                array('id', 'name'));
+                                array('id', 'name', 'parent_id'));
     
     $ret = array();
-    
     if($includeParent) {
+      //we must first check if there is/are parent(s) for this cou to include at the name
+      $parent_id = $parent['Cou']['parent_id'];
```
Comment (ioigoume) [inline app/Model/Cou.php:278]: $parent_id is redundant

**Review by ioigoume — CHANGES_REQUESTED** [review]

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**